vsintegration/src/FSharp.Editor/Navigation/NavigateToSearchService.fs

```diff
@@ -82,6 +82,22 @@ module private Index =
     type IIndexedNavigableItems =
         abstract Find: searchValue: string -> INavigateToSearchResult []
 
+    let private navigateToSearchResultComparer =
+        { new IEqualityComparer<INavigateToSearchResult> with 
+            member __.Equals(x: INavigateToSearchResult, y: INavigateToSearchResult) =
+                match x, y with
+                | null, _ | _, null -> false
+                | _ -> x.NavigableItem.Document.Id = y.NavigableItem.Document.Id &&
+                       x.NavigableItem.SourceSpan = y.NavigableItem.SourceSpan
+            
+            member __.GetHashCode(x: INavigateToSearchResult) =
+                if isNull x then 0
+                else
+                    let mutable hash = 17
+                    hash <- hash * 23 + x.NavigableItem.Document.Id.GetHashCode()
+                    hash <- hash * 23 + x.NavigableItem.SourceSpan.GetHashCode()
+                    hash }
```

Comment:
17 * 23 in the first line is always going to be the same number.  Probably the compiler will fold it.

How were these numbers chosen?  apart from being primes.

For my education could you please explain why a mutable field rather than:
````
    let docId = 17 * 23 + x.NavigableItem.Document.Id.GetHashCode() 
    docId * 23 + x.NavigableItem.SourceSpan.GetHashCode() 
````